kodi_addon_checker/schema_validation.py

```diff
@@ -0,0 +1,33 @@
+import xmlschema
+import logging
+
+from .report import Report
+from .record import PROBLEM, Record, INFORMATION
+
+xml_schema = "./xml_schema/"
+LOGGER = logging.getLogger(__name__)
+
+
+def schemas(report: Report, parsed_xml):
+
+    for extension in parsed_xml.findall("extension"):
+        extension_point = (extension.get("point")).split(".")[-1]
```

Comment:
you can't expect all provided schemas will be valid. The point attribute might not exist, this will crash your tool

Reply:
then we can either have a list to check if the point exists or not. Or we can use `try` and `expect`.

Reply:
yes and also other things:
- All addons MUST have a metadata extension element
- All addons MUST at least define one of the other extension elements

If the above conditions do not hold (plus the possibility of an incorrect `extension` element without the point attribute) you can fail the overall validation right away

Reply:
You mean add some early returns to this function? I disagree with that. The checker should always report all issues and not only some till you fix one.

Reply:
The comment was more directed towards introducing a few more validation steps to mzfr than deciding upon that. I agree the checker should report all problems even if an early one is found.